packages/ember-views/lib/components/component.js

```diff
@@ -456,6 +456,182 @@ var Component = View.extend(TargetActionSupport, {
+    and it also runs on subsequent re-renders, which is useful for
+    logic that is the same on all renders. It does not run when the
+    component has been re-rendered from the inside.
+
+    ```app/components/feedback-form.js
+    export default Ember.Component.extend({
+     didReceiveAttrs: function(options) {
+        this.set('title', 'Please provide us with your feedback');
+      }
+    });
+    ```
+
+    @public
+    @method didReceiveAttrs
+    @param options {Object}
+    @since 1.13.0
+  */
+
+
+  /**
+    `willRender` runs before the template is rendered. It runs when
+    the template is updated for any reason (both initial and
+    re-render, and regardless of whether the change was caused by an
+    attrs change or re-render)
+
+    ```app/components/todo-footer.js
+    export default Ember.Component.extend({
+      morethanOne: false,
+
+      willRender: function() {
+        if (this.get('items.length') > 1) {
+          this.set('morethanOne', true);
+        }
+      }
+    });
+    ```
+
+    @public
+    @method willRender
+    @since 1.13.0
+  */
+
+  /**
+   Called when the element of the component has been inserted into the DOM
+   or after the view was re-rendered. Override this function to do any set up
+   that requires an element in the document body.
+
+
+    `willRender` runs before the template is rendered. It runs when
+    the template is updated for any reason (both initial and
+    re-render, and regardless of whether the change was caused by an
+    attrs change or re-render)
+
+    ```app/components/edit-todo.js
+    export default Ember.Component.extend({
+      didInsertElement: function() {
+        this.$().focus();
+      }
+    });
+    ```
+
+    @public
+    @method didInsertElement
+  */
+
+  /**
+   `didRender` runs after `didInsertElement` (it also runs on
+   subsequent re-renders).
+
+    ```app/components/my-dropdown.js
+    import Ember from 'ember';
+    export default Ember.Component.extend({
+      didRender: function() {
+        Ember.$(document).updatePolyfill();
+      }
+    });
+    ```
+
+    @public
+    @method didRender
+    @since 1.13.0
+  */
+
+
+  /**
+   `didUpdateAttrs` runs when the attributes of a component have
+   changed (but not when the component is re-rendered, via
+   component.rerender, component.set, or changes in models or services
+   used by the template). It does not run on the first render.
+
+   The old attributes can be found on the `oldAttrs` property of the
+   options argument, and the new attributes can be found on the
+   `newAttrs` property.
+
+    ```app/components/book-table.js
+    export default Ember.Component.extend({
+      didUpdateAttrs: function(options) {
+        this.set(`selectedBooks`, []);
+      }
+    });
+    ```
+
+    @public
+    @method didUpdateAttrs
+    @param options {Object}
+    @since 1.13.0
+  */
+
+  /**
+   `willUpdate` runs when the component is re-rendering for any
```

Comment:
Should say "runs before the componetnt is re-rendered" to help make it clear where exactly this happens.